Documented 'hello world' sample does not work for Addons

### Version

v22.11.0+

### Platform

```text
Darwin LM-JNVH7RPW6J 23.6.0 Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu10063.141.2~1/RELEASE_ARM64_T6030 arm64
```


### Subsystem

_No response_

### What steps will reproduce the bug?

1. Go to https://nodejs.org/api/addons.html#hello-world
2. Copy the sample code into a `hello.cc` file
3. Try to build it with `node-gyp`.

### How often does it reproduce? Is there a required condition?

Always

### What is the expected behavior? Why is that the expected behavior?

N/A

### What do you see instead?

Build fails with 

```
../hello.cc:14:27: error: no matching member function for call to 'Set' 

args.GetReturnValue().Set(v8::String::NewFromUtf8(isolate, "world"));
```


### Additional information

_No response_

--- a/doc/api/addons.md
+++ b/doc/api/addons.md
@@ -72,6 +72,7 @@ namespace demo {
 using v8::FunctionCallbackInfo;
 using v8::Isolate;
 using v8::Local;
+using v8::NewStringType;
 using v8::Object;
 using v8::String;
 using v8::Value;
@@ -79,7 +80,7 @@ using v8::Value;
 void Method(const FunctionCallbackInfo<Value>& args) {
   Isolate* isolate = args.GetIsolate();
   args.GetReturnValue().Set(String::NewFromUtf8(
-      isolate, "world").ToLocalChecked());
+      isolate, "world", NewStringType::kNormal).ToLocalChecked());
 }
 
 void Initialize(Local<Object> exports) {

